src/hydro/srcterms/tabular_cooling.cpp

```diff
@@ -392,11 +393,18 @@ void TabularCooling::SubcyclingFixedIntSrcTerm(MeshData<Real> *md, const Real dt
 
             if (!dedt_valid) {
               if (sub_dt == min_sub_dt) {
-                PARTHENON_FAIL("FATAL ERROR in [TabularCooling::SubcyclingSplitSrcTerm]: "
-                               "Minumum sub_dt leads to negative internal energy");
+                //PARTHENON_FAIL("FATAL ERROR in [TabularCooling::SubcyclingSplitSrcTerm]: "
+                //               "Minumum sub_dt leads to negative internal energy");
+                
+                //HACK
+                //Even the minimum subcycle dt would lead to negative internal energy -- so just cool to the floor
```

Comment:
Is this physically correct, i.e., is the case where this is triggered always meaning that the cooling should go down to the floor or could it also be that this may happen higher up in the cooling table where there's just a bad subcycle (but the final temperature would not be at the floor)?
Also, if we've properly limited `dt` (say the default 10% fraction of change in internal energy) this should not be triggered or am I missing sth?

Reply:
This section of code will be triggered whenever a negative energy or NAN energy is encountered within any of the RK stages in the current subcycle and only when using the minimum subcycle timestep allowed.  I believe as long as there's a minimum timestep restriction on subcycles, this is the "last change" subcycle so not just any bad subcycle. 

This would lead to an incorrect final temperature if you had a cooling with a high cooling rate a few orders of magnitude above the temperature floor but a long asymptote to zero going down the the temperature flow. In practice though with the cooling tables I've been using, the cooling rate above the temperature floor is fast enough to drop to zero during this minimum subcycle timestep.

My best idea here is to check whether a subcycle at the current subcycle internal energy and minimum subcycle timestep would lead to a negative energy. If so, just cool to the floor. It might not be perfect, but I also don't have much interest trying to support  a contrived cooling table made to break the inferior cooling integrator when one can just use the `Townsend` integrator. 

You are also correct that this code is never encountered when `EstimateTimeStep` is enabled via `cooling/cfl=0.1` as you describe. During testing, however, I found that limiting the hydro timestep by the minimum cooling time led to the simulation to come to a halt when cold gas was forming. When I disabled the "cooling cfl", it lead to the commented out exception and my hack to get around it.

Reply:
This has been updated/addressed in #28, we'll need to merge those changes into this branch.